### DISCUSSION TITLE
Chapter 14 - Ecosystems

### DISCUSSION THREAD

**Discussion by argunaykut** [softmatterlab/SOCS#23] [General]
Here, you can ask questions and discuss issues about the simulations of Chapter 14 - Ecosystems. 
We encourage you to participate in these discussions and engage with our community.

**Comment by acallegar** [comment]
Hi, for the students asking about exercise 14.10

the relevant equations are (14.23) (system of dx/dt and dy/dt)
and (14.24) (equilibrium points; the first one is non-trivial)

--- a **possible** choice for the simulation parameters [they appear in equation (14.23)]

r1 = 0.01;
r2 = 0.02;
K1 = 140;
K2 = 150;
a12 = 0.5;
a21 = 0.7;

One way to solve 14.10 is:
plot the phase space, plot the equilibrium point, plot a few trajectories. Ex, see the figure below:


![Competition_phase_space](https://user-images.githubusercontent.com/66014271/205373304-15ae43c5-157f-4f55-8f96-9b6dc57e02e7.png)


The different colors represent different trajectories (they start from different initial x,y values).

See below for the time dependence of the trajectories x(t) and y(t)

black trajectory:

![Competition_traj_1](https://user-images.githubusercontent.com/66014271/205373715-c71166ae-34ce-4a4c-84c3-eb5e26428156.png)


blue trajectory:

![Competition_traj_2](https://user-images.githubusercontent.com/66014271/205373728-a99c995e-ec8f-459d-a883-0102928535fc.png)

**Comment by acallegar** [comment]
Hi, for the students asking about exercise 14.7

The finite difference equation for the explicit Euler is:

you take equation (14.15) and write
dx = x_(n+1) - x_(n)

and get
x_(n+1) = x_(n) + dt * (r * x_(n) - alpha * x_(n)^2)
